### PR TITLE
Fixed bug where storage class select wouldn't appear in NIM Modal

### PR DESCRIPTION
**Pull request by TheiaSurette**
## Description

Enhanced ManageNIMServingModal to support OpenShift default storage class and improve storage class selection logic

- Added support for OpenShift default storage class in ManageNIMServingModal.
- Updated logic to prefer ODH default storage class over OpenShift default when both are available.
- Introduced a new prop `showDefaultWhenNoConfig` in StorageClassSelect to control visibility and enablement of storage class options based on configuration availability.
- Added comprehensive tests for ManageNIMServingModal and StorageClassSelect to cover new functionality and edge cases.

Before: If no storage classes were defined in ODH, the storage class select would not appear and the model could not be deployed.
Now: If no storage classes are defined in ODH, it defaults to the OpenShift cluster default and displays it in a disabled dropdown.
## How Has This Been Tested?

I manually tested the behavior by removing the annotations from the storage cluster and trying to deploy a model. After the code changes, the OpenShift default is visible and the model can be deployed.

## Test Impact

I wrote 2 new unit test files to cover my changes as well as various edge cases

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [X] Included any necessary screenshots or gifs if it was a UI change.
image

- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

## Summary by CodeRabbit

* **New Features**
  * Improved storage class selection in deployment modals with fallback to OpenShift defaults when admin defaults are unavailable.
  * Added option to display all storage classes with selection disabled when no configurations exist, enhancing user clarity.

* **Bug Fixes**
  * Correctly disabled storage class selection during editing of existing deployments.

* **Tests**
  * Added comprehensive tests for storage class selection and deployment modal behavior, covering configuration states, form validation, and UI interactions.

### CODE REVIEW

File: frontend/src/pages/projects/screens/spawner/storage/StorageClassSelect.tsx [hunk 1]
```diff
@@ -129,7 +145,8 @@ const StorageClassSelect: React.FC<StorageClassSelectProps> = ({
         onChange={(selection) => {
           setStorageClassName(selection);
         }}
-        isDisabled={disableStorageClassSelect || !storageClassesLoaded}
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        isDisabled={disableStorageClassSelect || !storageClassesLoaded || shouldShowDefaultOnly}
```
Comment (emilys314) [inline frontend/src/pages/projects/screens/spawner/storage/StorageClassSelect.tsx:149]: We should make this work without the `eslint-disable` because on line 137 you are asserting it can only be truthy. Typescript is correctly picking this issue up

File: frontend/src/pages/modelServing/screens/projects/NIMServiceModal/ManageNIMServingModal.tsx [hunk 2]
```diff
@@ -147,8 +148,11 @@ const ManageNIMServingModal: React.FC<ManageNIMServingModalProps> = ({
   );
 
   const isStorageClassesAvailable = useIsAreaAvailable(SupportedArea.STORAGE_CLASSES).status;
-  const [defaultSc] = useAdminDefaultStorageClass();
-  const defaultStorageClassName = defaultSc?.metadata.name || '';
+  const odhDefaultScResult = useAdminDefaultStorageClass();
+  const openshiftDefaultSc = useOpenshiftDefaultStorageClass();
+  const odhDefaultSc = odhDefaultScResult[0];
+  const defaultSc = odhDefaultSc || openshiftDefaultSc;
+  const defaultStorageClassName = defaultSc ? defaultSc.metadata.name || '' : '';
```
Comment (emilys314) [inline frontend/src/pages/modelServing/screens/projects/NIMServiceModal/ManageNIMServingModal.tsx:155]: Very minor nitpick but the ternary seems a little redundant 
```suggestion
  const defaultStorageClassName = defaultSc?.metadata.name || '';
```

(Also technically you only have to fall back to the empty string when it's nullish so i think a `??` is more accurate to have)
```suggestion
  const defaultStorageClassName = defaultSc?.metadata.name ?? '';
```

**Review by emilys314 — COMMENTED** [review]
Nice that's a lot of tests added

**Comment by emilys314** [comment]
looks good
/lgtm
/approve

image

**Review by emilys314 — APPROVED** [review]

**Comment by andrewballantyne** [comment]
Merging manually as Konflux is blocking the merge.
